lib/Controller/AuthorizationController.php

```diff
@@ -0,0 +1,151 @@
+ * and https://tools.ietf.org/rfc/rfc7231.txt and https://tools.ietf.org/rfc/rfc7235.txt
+ * for an in-depth discussions on what HTTP reponse headers to use. In short:
+ *
+ *		401 UNAUTHORIZED	Indicates that the request has not been applied because it lacks
+ * 							valid authentication credentials for the target resource.
+ *							Response MUST send a WWW-Authenticate header field.
+ *
+ *		403	FORBIDDEN		Indicates that the server understood the request but refuses to authorize it.
+ *							If authentication credentials were provided in the request, the
+ *							server considers them insufficient to grant access.
+ *
+ * @author Andreas Goetz <cpuidle@gmx.de>
+ * @package default
+ */
+class AuthorizationController extends Controller {
+
+	const TOKEN_LIFETIME = 24 * 3600;
+
+	/**
+	 * Authorize request via header
+	 */
+	public static function authorize(Request $request, View $view) {
+		if (!($key = Util\Configuration::read('authorization.secretkey'))) {
+			$view->getResponse()->setStatusCode(Response::HTTP_INTERNAL_SERVER_ERROR);
+			throw new \Exception('Missing authorization.secretkey');
+		}
+
+		try {
+			if (!($header = $request->headers->get('Authorization')) || (0 !== strpos($header, 'Bearer '))) {
+				throw new \Exception('Missing authorization token');
+			}
+
+			$jwt = substr($header, strlen('Bearer '));
+			if ($token = JWT::decode($jwt, $key, array('HS256'))) {
+				// Check if this token has expired due to changed config
+				if (isset($token->iat) && ($token->iat + Util\Configuration::read('authorization.valid', self::TOKEN_LIFETIME)) < time()) {
+					throw new ExpiredException('Expired token');
+				}
+			}
+		}
+		catch (\Exception $e) {
+			$view->getResponse()->headers->set('WWW-Authenticate', 'Bearer');
+			$view->getResponse()->setStatusCode(Response::HTTP_UNAUTHORIZED);
+			throw($e);
+		}
+	}
+
+	/**
+	 * Run operation
+	 *
+	 * @param string $operation runs the operation if class method is available
+	 */
+	public function add() {
+		try {
+			$json = Util\JSON::decode($this->request->getContent());
+		}
+		catch (\Exception $e) {
+			$this->view->getResponse()->setStatusCode(Response::HTTP_FORBIDDEN);
+			throw new \Exception('Invalid token request');
+		}
+/*
+		// firebase token auth
+		if (isset($json->authtype) && $json->authtype == 'firebase' && isset($json->authtoken)) {
+			$authtoken = trim($json->authtoken);
+
+			$googlePublicKeys = json_decode(file_get_contents(VZ_DIR . '/etc/firebase_public.json'), true);
+			$token = JWT::decode($authtoken, $googlePublicKeys, ['RS256']);
+
+			$auth = Util\Configuration::read('users.firebase');
+
+			if (in_array($token->email, $auth)) {
+				$jwt = $this->issueToken($token->email);
+				return(['authtoken' => $jwt]);
+			}
+		}
+		else
+*/
+		// username/password
+		if (isset($json->username) && isset($json->password)) {
+			$username = strtolower(trim($json->username));
+			$password = trim($json->password);
```

Comment:
I do not know the exact password rules, but if spaces are allowed this may lead to problems.